src/test/java/org/jenkinsci/plugins/workflow/actions/ErrorActionTest.java

```diff
@@ -243,4 +349,15 @@ public static class X extends Exception {
         assertNotNull(new ErrorAction(unserializable));
         assertNotNull(new ErrorAction(cyclic));
     }
+
+    @Test public void findOriginOfRethrownUnserializableException() throws Throwable {
+        rr.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                "stage('test') { withEnv([]) {  throw new " + X.class.getCanonicalName() + "() } }", false /* for "new org.jenkinsci.plugins.workflow.actions.ErrorActionTest$X" */));
+            WorkflowRun b = r.buildAndAssertStatus(Result.FAILURE, p);
+            FlowNode originNode = ErrorAction.findOrigin(b.getExecution().getCauseOfFailure(), b.getExecution());
+            assertThat(((StepNode) originNode).getDescriptor().getFunctionName(), equalTo("withEnv"));
+        });
+    }
```

Comment:
This test is weak, it only passed prior to https://github.com/jenkinsci/workflow-api-plugin/pull/328/commits/15024e3bc079c4fcc9de6c20758acaaa17fb8575 because of the stack trace equality fallback in `Object.equals`. I would probably delete it in favor of the new test.

```suggestion
```